mcc/odin/odin/mock_rds_client_test.go

```diff
@@ -175,6 +175,43 @@ func (m *mockRDSClient) AddSnapshots(
 	)
 }
 
+// CreateDBSnapshot mocks rds.CreateDBSnapshot.
+func (m *mockRDSClient) CreateDBSnapshot(
+	params *rds.CreateDBSnapshotInput,
+) (
+	output *rds.CreateDBSnapshotOutput,
+	err error,
+) {
+	instanceID := params.DBInstanceIdentifier
+	_, instance, err := m.FindInstance(*instanceID)
+	if err != nil {
+		return
+	}
+	if *instance.DBInstanceStatus != "available" {
+		err = fmt.Errorf(
+			"%s instance state is not available",
+			instanceID,
+		)
+	}
+	id := params.DBSnapshotIdentifier
+	_, snapshot, _ := m.FindSnapshot(*id)
+	if snapshot != nil {
```

Comment:
If `FindSnapshot` returns a `nil` snapshot, this means, it was not found.
Having this check, we take the happy path opposite than usual.